core/ipv6: fix warnings for building G3 and improve RFC conformance

julianuziemblo opened the pull request:
JIRA: RTOS-1098

julianuziemblo committed:
core/ipv6: fix warnings

JIRA: RTOS-1098

nalajcie commented on src/core/ipv6/ip6_addr.c:
```diff
@@ -266,7 +266,7 @@ ip6addr_ntoa_r(const ip6_addr_t *addr, char *buf, int buflen)
         break;
       }
-      if (empty_block_flag == 0) {
+      if (empty_block_flag == 0 && ((current_block_index + 1) >> 1) < LWIP_ARRAYSIZE(addr->addr)) {
```
IMHO the correct way to handle it is to remove `&& empty_block_flag == 1` from line 261.

Quick google search confirms: https://gitee.com/src-openeuler/lwip/blob/master/0171-IPV6-fix-array-out-of-bounds-in-function-ip6_ntoa_r.patch

Please analyze & test & confirm.

julianuziemblo replied:
I don't understand the link you sent - it fixes the error in exactly the same way as I do. However,  your proposed fix would be indeed cleaner, so I'll analyze and try it

julianuziemblo replied:
Tested it, removing `&& empty_block_flag == 1`  from line 261 makes address lose its last 0 if there was an empty block before ("::"), e.g. `2001:db8::2:0` renders as `2001:db8::2:`, so this is not a valid solution.

julianuziemblo replied:
ebd103775dcf1285c0eca87f83a0c48d0922da8b adds `&& empty_block_flag == 1` to line 261 describing this exact point

nalajcie replied:
yeah, the proposed fix was invalid for corner case. We decided to use simpler condition to align with existing code more seamlessly.

julianuziemblo committed:
core/ipv6: conform to RFC 5952

RFC 5952 requires that all IPv6 addresses with hex
characters "a" through "f" are represented using
lowercase letters.

JIRA: RTOS-1098
